src/main/java/TaskList.java

```diff
@@ -1,4 +1,75 @@
-package PACKAGE_NAME;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
 
 public class TaskList {
+
+    private static DateTimeFormatter dukeDateTimeFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    private ArrayList<Task> dukeTaskList = null;
+
+    public TaskList(ArrayList<Task> dukeTaskList) {
+        this.dukeTaskList = dukeTaskList;
+    }
+
+    public TaskList() {
+        this.dukeTaskList = new ArrayList<>();
+    }
+
+    public void addTask(Task toAdd) {
+        dukeTaskList.add(toAdd);
+    }
+
+    public void deleteTask(Task toDelete) {
+        dukeTaskList.remove(toDelete);
+    }
+
+    public Task getTask(int index) {
+        return dukeTaskList.get(index);
+    }
+
+    public Task removeTask(int index) {
+        return dukeTaskList.remove(index);
+    }
+
+    public ArrayList<Task> getDukeTaskList() {
+        return dukeTaskList;
+    }
+
+    public int getSize() {
+        return dukeTaskList.size();
+    }
+
+    public void addTaskAfterValidation(String dateTimeString, Task t) {
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(dateTimeString, dukeDateTimeFormatter);
+
+            if (t instanceof Event) {
+                Event e = (Event) t;
+                e.setDateTimeAt(dateTime);
+            } else {
```

Comment:
Perhaps it could be better to write "else if ( t instanceof Deadline) { ... } " and use the else-block for catching exceptions etc, because according to coding standards the final else "should mean everything else", and not the final option!